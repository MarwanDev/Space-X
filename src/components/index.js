export { default as Navbar } from './Navbar/Navbar';
export { default as Rockets } from './Rockets/Rockets';
export { default as Missions } from './Missions/Missions';
export { default as MyProfile } from './MyProfile/MyProfile';

