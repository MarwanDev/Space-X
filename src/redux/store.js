import { configureStore } from "@reduxjs/toolkit";
import rocketReducer from "./RocketSlice/rocketsSlice";


const store = configureStore({
  reducer: {
    rockets: rocketReducer,
  },

});

export default store;
