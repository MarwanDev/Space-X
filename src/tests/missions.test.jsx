import { expect, it, describe } from "vitest";
import renderer from "react-test-renderer";
import { render } from '@testing-library/react';
import store from "../redux/store";
import { Provider } from "react-redux";
import { Missions } from "../components";
import reducer, {
  joinMission,
  cancelMission,
} from "../redux/missionSlice/missionsSlice";

describe("Missions", () => {
  it("should reserve missions correctly", () => {
    const state = {
      missions: [
        {
          mission_id: "0",
          mission_name: "Mission 1",
          description: "Mission 1",
        },
        {
          mission_id: "1",
          mission_name: "Mission 2",
          description: "Mission 2",
        },
        {
          mission_id: "2",
          mission_name: "Mission 3",
          description: "Mission 3",
        },
      ],
      Loading: false,
    };
    expect(reducer(state, joinMission("1"))).toEqual({
      missions: [
        {
          mission_id: "0",
          mission_name: "Mission 1",
          description: "Mission 1",
        },
        {
          mission_id: "1",
          mission_name: "Mission 2",
          description: "Mission 2",
          joined: true,
        },
        {
          mission_id: "2",
          mission_name: "Mission 3",
          description: "Mission 3",
        },
      ],
      Loading: false,
    });
  });

  it("should cancel missions reservation correctly", () => {
    const state = {
      missions: [
        {
          mission_id: "0",
          mission_name: "mission 1",
          description: "mission 1",
          joined: true,
        },
        {
          mission_id: "1",
          mission_name: "mission 2",
          description: "mission 2",
          joined: true,
        },
        {
          mission_id: "2",
          mission_name: "mission 3",
          description: "mission 3",
          joined: true,
        },
      ],
      isLoading: false,
    };
    expect(reducer(state, cancelMission("1"))).toEqual({
      missions: [
        {
          mission_id: "0",
          mission_name: "mission 1",
          description: "mission 1",
          joined: true,
        },
        {
          mission_id: "1",
          mission_name: "mission 2",
          description: "mission 2",
          joined: false,
        },
        {
          mission_id: "2",
          mission_name: "mission 3",
          description: "mission 3",
          joined: true,
        },
      ],
      isLoading: false,
    });
  });
  it("renders missions correctly", () => {
    const missions = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>
      )
      .toJSON();
    expect(missions).toMatchSnapshot();
  });
  it("renders missions correctly", () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>
    );
  });
});
