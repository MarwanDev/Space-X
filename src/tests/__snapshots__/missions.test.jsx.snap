// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Missions > renders missions correctly 1`] = `
<div
  className="table-container"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root table css-rqglhn-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-duongd-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1nl57la-MuiTableCell-root"
            scope="col"
          >
            Mission
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium mission-desc css-1nl57la-MuiTableCell-root"
            scope="col"
          >
            Description
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium mission-status css-1nl57la-MuiTableCell-root"
            scope="col"
          >
            Status
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium mission-option css-1nl57la-MuiTableCell-root"
            scope="col"
          />
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      />
    </table>
  </div>
</div>
`;
